### ISSUE TITLE
Cannot successfully install gtk3 gem

### ISSUE THREAD

**Issue by soumyaray** [ruby-gnome2/ruby-gnome2#744]
Several gems failed to build trying to bundle install gtk3 in my Gemfile.  Finally, looking at [this page](https://github.com/ruby-gnome2/ruby-gnome2/issues/259), I found I had to put `PKG_CONFIG_PATH=/usr/local/opt/libffi/lib/pkgconfig gem install ...` for each failing gem dependency.

I think doing:

    PKG_CONFIG_PATH=/usr/local/opt/libffi/lib/pkgconfig bundle install

would have worked just  as well.

Everything installs now. But when i try to `require 'gtk3'` in pry/irb, I get:

    NoMethodError: undefined method `prepend_dll_path' for GLib:Module
    from /Users/username/.rvm/gems/ruby-2.2.3/gems/gobject-introspection-3.0.8/lib/gobject-introspection.rb:37:in `&lt;top (required)&gt;'

**Comment by cedlemo** [comment]
Hi @soumyaray,
is possible to know the steps you follow to install gtk3 and the failing gems (which one failed, which order).

could you try this in your pry/irb:

```
require "glib2"
GLib.methods.grep(/prepend/)
GLib.methods 
```

I am surprised that gobject-introspection.rb can require Glib2 but cannot find the prepend_dll_path method.

**Comment by kou** [comment]
Could you try the following?

```text
% curl -O http://pub.cozmixng.org/~kou/tmp/glib2-3.0.8.gem
% gem install glib2-3.0.8.gem
% gem install gobject-introspection # <- No PKG_CONFIG_PATH
% gem install gtk3                           # <- No PKG_CONFIG_PATH
```

The `glib2-3.0.8.gem` has workaround for the `libffi.pc` problem. But I didn't test it because I don't have OS X. Could you help us to fix this problem?

**Comment by kou** [comment]
Could you try updating your RubyGems for the `prepend_dll_path` problem?

**Comment by soumyaray** [comment]
Hi @cedlemo: Sorry I can't recall all the steps I followed :(  i started with trying to install the gtk3 gem.  I think I had to install another unix package first but don't recall which one. After that, I've put all the installation steps I recall in the issue description.

Here's the results of the pry steps you asked about:
```
[1] pry(main)> require "glib2"
=> true
[2] pry(main)> GLib.methods.grep(/prepend/)
=> [:prepend_path_to_environment_variable, :prepend_dll_path, :prepend]
[3] pry(main)> GLib.methods
=> [:check_binding_version?,
 :exit_application,
 :__add_one_arg_setter,
 :prepend_path_to_environment_variable,
 :prepend_dll_path,
 :os_win32?,
 :os_beos?,
 :os_unix?,
 :utf8_validate,
 :convert,
 :locale_to_utf8,
 :locale_from_utf8,
 :filename_to_utf8,
 :filename_from_utf8,
 :filename_to_uri,
 :filename_from_uri,
 :format_size_for_display,
 :format_size,
 :language_names,
 :application_name,
 :set_application_name,
 :application_name=,
 :prgname,
 :set_prgname,
 :prgname=,
 :getenv,
 :setenv,
 :unsetenv,
 :listenv,
 :host_name,
 :user_name,
 :real_name,
 :user_cache_dir,
 :user_data_dir,
 :user_config_dir,
 :system_data_dirs,
 :system_config_dirs,
 :get_user_special_dir,
 :home_dir,
 :tmp_dir,
 :current_dir,
 :path_is_absolute?,
 :path_skip_root,
 :path_get_basename,
 :path_get_dirname,
 :find_program_in_path,
 :bit_nth_lsf,
 :bit_nth_msf,
 :bit_storage,
 :spaced_primes_closest,
 :parse_debug_string,
 :check_version?,
 :set_ruby_thread_priority,
 :ruby_thread_priority=,
 :charset,
 :freeze,
 :===,
 :==,
 :<=>,
 :<,
 :<=,
 :>,
 :>=,
 :to_s,
 :inspect,
 :included_modules,
 :include?,
 :name,
 :ancestors,
 :instance_methods,
 :public_instance_methods,
 :protected_instance_methods,
 :private_instance_methods,
 :constants,
 :const_get,
 :const_set,
 :const_defined?,
 :const_missing,
 :class_variables,
 :remove_class_variable,
 :class_variable_get,
 :class_variable_set,
 :class_variable_defined?,
 :public_constant,
 :private_constant,
 :singleton_class?,
 :include,
 :prepend,
 :module_exec,
 :class_exec,
 :module_eval,
 :class_eval,
 :method_defined?,
 :public_method_defined?,
 :private_method_defined?,
 :protected_method_defined?,
 :public_class_method,
 :private_class_method,
 :autoload,
 :autoload?,
 :instance_method,
 :public_instance_method,
 :pretty_print_cycle,
 :pretty_print,
 :pry,
 :__binding__,
 :pretty_print_instance_variables,
 :pretty_print_inspect,
 :nil?,
 :=~,
 :!~,
 :eql?,
 :hash,
 :class,
 :singleton_class,
 :clone,
 :dup,
 :itself,
 :taint,
 :tainted?,
 :untaint,
 :untrust,
 :untrusted?,
 :trust,
 :frozen?,
 :methods,
 :singleton_methods,
 :protected_methods,
 :private_methods,
 :public_methods,
 :instance_variables,
 :instance_variable_get,
 :instance_variable_set,
 :instance_variable_defined?,
 :remove_instance_variable,
 :instance_of?,
 :kind_of?,
 :is_a?,
 :tap,
 :send,
 :public_send,
 :respond_to?,
 :extend,
 :display,
 :method,
 :public_method,
 :singleton_method,
 :define_singleton_method,
 :object_id,
 :to_enum,
 :enum_for,
 :pretty_inspect,
 :equal?,
 :!,
 :!=,
 :instance_eval,
 :instance_exec,
 :__send__,
 :__id__]
```

**Comment by soumyaray** [comment]
@kou: I was able to download and install the `glib` gem, but things things went wrong with `gobject-introspection`:

```
$ gem install gobject-introspection
Building native extensions.  This could take a while...
ERROR:  Error installing gobject-introspection:
	ERROR: Failed to build gem native extension.

    /Users/username/.rvm/rubies/ruby-2.2.3/bin/ruby -r ./siteconf20160409-24143-1l9hjac.rb extconf.rb
checking for --enable-debug-build option... no
checking for -Wall option to compiler... yes
checking for -Waggregate-return option to compiler... yes
checking for -Wcast-align option to compiler... yes
checking for -Wextra option to compiler... yes
checking for -Wformat=2 option to compiler... yes
checking for -Winit-self option to compiler... yes
checking for -Wlarger-than-65500 option to compiler... yes
checking for -Wmissing-declarations option to compiler... yes
checking for -Wmissing-format-attribute option to compiler... yes
checking for -Wmissing-include-dirs option to compiler... yes
checking for -Wmissing-noreturn option to compiler... yes
checking for -Wmissing-prototypes option to compiler... yes
checking for -Wnested-externs option to compiler... yes
checking for -Wold-style-definition option to compiler... yes
checking for -Wpacked option to compiler... yes
checking for -Wp,-D_FORTIFY_SOURCE=2 option to compiler... yes
checking for -Wpointer-arith option to compiler... yes
checking for -Wswitch-default option to compiler... yes
checking for -Wswitch-enum option to compiler... yes
checking for -Wundef option to compiler... yes
checking for -Wout-of-line-declaration option to compiler... yes
checking for -Wunsafe-loop-optimizations option to compiler... no
checking for -Wwrite-strings option to compiler... yes
checking for rb_define_alloc_func() in ruby.h... yes
checking for rb_block_proc() in ruby.h... yes
checking for new allocation framework... yes
checking for attribute assignment... no
checking for Windows... no
checking for gobject-introspection-1.0... yes
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/Users/username/.rvm/rubies/ruby-2.2.3/bin/$(RUBY_BASE_NAME)
	--enable-debug-build
	--disable-debug-build
	--with-pkg-config
	--without-pkg-config
	--with-override-variables
	--without-override-variables
/Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:265:in `parse_pc': .pc for libffi doesn't exist. (RuntimeError)
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:185:in `declaration'
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:199:in `collect_cflags'
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:138:in `cflags'
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:201:in `block in collect_cflags'
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:200:in `collect'
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:200:in `collect_cflags'
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:138:in `cflags'
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:395:in `cflags'
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/pkg-config-1.1.7/lib/pkg-config.rb:446:in `have_package'
	from /Users/username/.rvm/gems/ruby-2.2.3/gems/glib2-3.0.8/lib/mkmf-gnome2.rb:620:in `required_pkg_config_package'
	from extconf.rb:59:in `<main>'

extconf failed, exit code 1

Gem files will remain installed in /Users/username/.rvm/gems/ruby-2.2.3/gems/gobject-introspection-3.0.8 for inspection.
Results logged to /Users/username/.rvm/gems/ruby-2.2.3/extensions/x86_64-darwin-14/2.2.0-static/gobject-introspection-3.0.8/gem_make.out
```

**Comment by soumyaray** [comment]
@kou: can you help me with some more details about what to do for this?

> Could you try updating your RubyGems for the prepend_dll_path problem?

**Comment by kou** [comment]
Sorry. I uploaded old gem. Could you try them again?

```text
% curl -O http://pub.cozmixng.org/~kou/tmp/glib2-3.0.8.2016-04-10.gem
% gem install glib2-3.0.8.2016-04-10.gem
% gem install gobject-introspection # <- No PKG_CONFIG_PATH
% gem install gtk3                           # <- No PKG_CONFIG_PATH
```

**Comment by kou** [comment]
> @kou: can you help me with some more details about what to do for this?
> 
> > Could you try updating your RubyGems for the prepend_dll_path problem?

See https://rubygems.org/pages/download

**Comment by kou** [comment]
This will be fixed in 3.0.9.
